rush.json

```diff
@@ -1,20 +1,300 @@
+   * correct error-underlining and tab-completion for editors such as VS Code.
+   */
   "rushVersion": "5.5.2",
+
+  /**
+   * The next field selects which package manager should be installed and determines its version.
+   * Rush installs its own local copy of the package manager to ensure that your build process
+   * is fully isolated from whatever tools are present in the local environment.
+   *
+   * Specify one of: "pnpmVersion", "npmVersion", or "yarnVersion".  See the Rush documentation
+   * for details about these alternatives.
+   */
+  "pnpmVersion": "2.19.2",
+
+  // "npmVersion": "5.4.0",
+  // "yarnVersion": "1.9.4",
+
+  /**
+   * Options that are only used when the PNPM package manager is selected
+   */
+  "pnpmOptions": {
+    /**
+     * If true, then Rush will add the "--strict-peer-dependencies" option when invoking PNPM.
+     * This causes "rush install" to fail if there are unsatisfied peer dependencies, which is
+     * an invalid state that can cause build failures or incompatible dependency versions.
+     * (For historical reasons, JavaScript package managers generally do not treat this invalid
+     * state as an error.)
+     *
+     * The default value is false to avoid legacy compatibility issues.
+     * It is strongly recommended to set strictPeerDependencies=true.
+     */
+    "strictPeerDependencies": true
+  },
+
   "repository": {
+    /**
+     * This setting is sometimes needed when using "rush change" with multiple Git remotes.
+     * It specifies the remote url for the official Git repository.  If this URL is provided,
+     * "rush change" will use it to find the right remote to compare against.
+     */
     "url": "https://github.com/OfficeDev/office-ui-fabric-react.git"
   },
+
+  /**
+   * Event hooks are customized script actions that Rush executes when specific events occur
+   */
   "eventHooks": {
-    "postRushInstall": ["node scripts/install-husky.js"]
+    /**
+     * The list of shell commands to run before the Rush installation starts
+     */
+    "preRushInstall": [
+      // "common/scripts/pre-rush-install.js"
+    ],
+
+    /**
+     * The list of shell commands to run after the Rush installation finishes
+     */
+    "postRushInstall": ["node scripts/install-husky.js"],
+
+    /**
+     * The list of shell commands to run before the Rush build command starts
+     */
+    "preRushBuild": [],
+
+    /**
+     * The list of shell commands to run after the Rush build command finishes
+     */
+    "postRushBuild": []
+  },
+
+  /**
+   * Older releases of the NodeJS engine may be missing features required by your system.
+   * Other releases may have bugs.  In particular, the "latest" version will not be a
+   * Long Term Support (LTS) version and is likely to have regressions.
+   *
+   * Specify a SemVer range to ensure developers use a NodeJS version that is appropriate
+   * for your repo.
+   */
+  "nodeSupportedVersionRange": ">=8.0.0 <10.0.0",
```

Comment:
><10.0.0 [](start = 40, length = 7)

<11.0.0?

Reply:
No I haven't tested Node 10 yet. I intend to do that in a separate PR.

Reply:
You can use `"nodeSupportedVersionRange": ">=8.9.4 <9.0.0 || >=10.13.0 <11.0.0"` to filter the LTS versions.

---
In reply to: [236400657](https://github.com/OfficeDev/office-ui-fabric-react/pull/7186#discussion_r236400657) [](ancestors = 236400657)